Loop/Managers/LoopDataManager.swift

```diff
@@ -1036,13 +1045,22 @@ extension LoopDataManager {
 
                     retrospectiveGlucoseEffect = computeRetrospectiveGlucoseEffect(startingAt: glucose, carbEffects: potentialCarbEffect)
                 }
-            } else {
+            } else if let carbEffect = carbEffectOverride ?? self.carbEffect {
                 effects.append(carbEffect)
             }
         }
 
-        if inputs.contains(.insulin), let insulinEffect = includingPendingInsulin ? self.insulinEffectIncludingPendingInsulin : self.insulinEffect {
-            effects.append(insulinEffect)
+        let computationInsulinEffect: [GlucoseEffect]?
```

Comment:
[nit] Consider putting lines 1053-1058 inside of block starting at line 1060 (`if inputs.contains(.insulin)`). The `computationInsulinEffect` is only used inside that block and it is unnecessary work if `.insulin` is not included in the `inputs`.

Reply:
In fact, I think you might be able to simplify this block with lines 1061-1063.